### PR TITLE
Refactors TransactionBuilder so that it does not depend on Wallet type

### PR DESCRIPTION
**Pull request by mtlynch**
Adds a new interface walletSpender, which wraps a subset of the Wallet type's
functionality. This allows us to unit test TransactionBuilder without a
dependency on the Wallet type. It will be possible to test TransactionBuilder
using a mock that implements walletSpender.

### CODE REVIEW

File: modules/wallet/transactionbuilder.go [hunk 1]
```diff
@@ -17,7 +108,7 @@ type transactionBuilder struct {
 	siacoinInputs []int
 	siafundInputs []int
 
-	wallet *Wallet
+	wallet walletSpender
```
Comment (DavidVorick) [inline modules/wallet/transactionbuilder.go]: not that it's a huge deal, but the name probably shouldn't be 'wallet' since it's actually a walletSpender.

**Comment by DavidVorick** [comment]
ok, just so we're clear on some things that have to happen at some point, even if not this PR:

+ interface should contain only exported functions
+ transaction builder should be split into a separate package
+ maps and other internal state cannot be exposed by the interface, but especially not maps
+ locks cannot be exposed by the interface, interface must be thread-safe without exposed locks

**Comment by DavidVorick** [comment]
other than the name 'wallet', I think this is ready to be merged into tbuilder

**Comment by mtlynch** [comment]
Cool, fixed thanks for the review.

**Comment by DavidVorick** [comment]
Luke and I discussed, if the transaction builder gets moved into its own package, it should be a subpackage of the wallet

**Pull request merged**